packages/runtime/tests/graphos.test.ts

```diff
@@ -0,0 +1,195 @@
+import { setTimeout } from 'timers/promises';
+import {
+  DefaultLogger,
+  type GatewayConfigContext,
+  type GatewayGraphOSManagedFederationOptions,
+} from '@graphql-hive/gateway';
+import { Response } from '@whatwg-node/fetch';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { createGraphOSFetcher } from '../src/fetchers/graphos';
+
+describe('GraphOS', () => {
+  describe('supergraph fetching', () => {
+    vi.useFakeTimers?.();
+    const advanceTimersByTimeAsync = vi.advanceTimersByTimeAsync || setTimeout;
+
+    beforeEach(() => {
+      vi.clearAllMocks();
+    });
+
+    it('should fetch the supergraph SDL', async () => {
+      const { unifiedGraphFetcher } = createTestFetcher({ fetch: mockSDL });
+
+      const result = unifiedGraphFetcher({});
+      await advanceTimersByTimeAsync(1_000);
+      expect(await result).toBe(supergraphSdl);
+    });
+
+    it('should retry on error', async () => {
+      let tries = 0;
+      const { unifiedGraphFetcher } = createTestFetcher({
+        fetch: () => {
+          tries++;
+          if (tries === 1) {
+            return mockFetchError();
+          }
+          return mockSDL();
+        },
+      });
+
+      const result = unifiedGraphFetcher({});
+      for (let i = 0; i < 3; i++) {
+        await advanceTimersByTimeAsync(1_000);
+      }
+
+      expect(await result).toBe(supergraphSdl);
+    });
+
+    it('should not retry more than maxRetry', async () => {
+      const { unifiedGraphFetcher } = createTestFetcher(
+        { fetch: mockFetchError },
+        { maxRetries: 3 },
+      );
+
+      const result = unifiedGraphFetcher({}).catch((err) => err);
+      for (let i = 0; i < 3; i++) {
+        await advanceTimersByTimeAsync(1_000);
+      }
+      expect(await result).toBeInstanceOf(Error);
```

Comment:
Isn't that weird ? It should not resolve to an error ? It should throw and error ?

Reply:
I made it resolve the error in the catch clause;
https://github.com/graphql-hive/gateway/pull/447/files#diff-6f0ac42f390077837143cf368831d9825e51dbba58dcb9b4e6eb77d4cd3cd5f4R54
Because Bun tester throws the error instead of catching it with `expect`